sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2454,6 +2472,25 @@ def setup_docker_user(self, cluster_config_file: str):
                                                     cluster_config_file)
         self.docker_user = docker_user
 
+    def setup_vpn(self, vpn_config: vpn_utils.VPNConfig) -> None:
+        assert self.cached_cluster_info is not None, (
+            'cluster_info should be set before setting up VPN.')
+        instance_ids = self.cached_cluster_info.instance_ids()
+        runners = self.get_command_runners()
+
+        def _run_setup_commands(runner_instance_id):
+            runner, instance_id = runner_instance_id
+            command = vpn_config.get_setup_commands(instance_id)
```

Comment:
Using instance id as hostname might not be the most intuitive choice, as user might get confused about a bunch of `i-shjadhak` on the tailscale console. How about `sky-{cluster_name}-{user_hash}-{node_id}`?

Reply:
Lets use this function to keep the length inside the limit

https://github.com/skypilot-org/skypilot/blob/3f898abe10c1aa2a05da0e00f0c6a9a947b53bbc/sky/utils/common_utils.py#L156-L158